daemon/algod/api/server/v2/handlers.go

```diff
@@ -364,6 +319,68 @@ func (v2 *Handlers) RawTransaction(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
 }
 
+// TealDryRun takes transactions and additional simulated ledger state and returns debugging information.
+// (POST /v2/teal/dryrun)
+func (v2 *Handlers) TealDryRun(ctx echo.Context) error {
+	if !v2.Node.Config().EnableDeveloperAPI {
+		return ctx.String(http.StatusNotFound, "/teal/dryrun was not enabled in the configuration file by setting the EnableDeveloperAPI to true")
+	}
+	req := ctx.Request()
+	buf := new(bytes.Buffer)
+	req.Body = http.MaxBytesReader(nil, req.Body, maxTealDryrunBytes)
+	buf.ReadFrom(req.Body)
+	data := buf.Bytes()
+
+	var dr DryrunRequest
+	var gdr generated.DryrunRequest
+	err := decode(protocol.JSONHandle, data, &gdr)
```

Comment:
The decoding here need to be determined by Content-Type and not by trying and failing.

Reply:
@algorandskiy

Reply:
I agree in general. But I also find annoying setting content-type from `curl`, and this endpoint more for developers so this is a bit of convenience.